### PR TITLE
Change APIs to differentiate hardware and software breakpoints

### PR DESCRIPTION
**Pull request by fjricci**

### CODE REVIEW

File: Headers/DebugServer2/Target/ProcessBase.h [hunk 1]
```diff
@@ -13,14 +13,14 @@
...
+#include "DebugServer2/SoftwareBreakpointManager.h"
 
 #include <functional>
 #include <set>
 
 namespace ds2 {
 
-class BreakpointManager;
-class WatchpointManager;
+class HardwareBreakpointManager;
```
Comment (sas) [inline Headers/DebugServer2/Target/ProcessBase.h]: You probably wanna remove this as well, and include the proper header after line 16 (`HardwareBreakpointManager.h`).

Comment (fjricci) [inline Headers/DebugServer2/Target/ProcessBase.h]: That header doesn't exist yet (just like WatchpointManager never did before)

Comment (sas) [inline Headers/DebugServer2/Target/ProcessBase.h]: Yeah but you should still remove that forward declaration then :)

Comment (fjricci) [inline Headers/DebugServer2/Target/ProcessBase.h]: Well it's needed for the API. I could do more in this commit, but it was really intended as just a "rename the api to be more accurate" commit.

**Comment by sas** [comment]
@facebook-github-bot shipit

I saw a couple of unsorted headers, but besides that this PR is good.

**Comment by sas** [comment]
@facebook-github-bot shipit

**Comment by fjricci** [comment]
Committed as 921c072c37376528e28e5897fcb8475bd8a77a81

**Pull request closed without merging**